diff --git a/node_modules/osu-parser/lib/slidercalc.js b/node_modules/osu-parser/lib/slidercalc.js
index 97c710b..d2654db 100644
--- a/node_modules/osu-parser/lib/slidercalc.js
+++ b/node_modules/osu-parser/lib/slidercalc.js
@@ -65,6 +65,10 @@ exports.getEndPoint = function (sliderType, sliderLength, points) {
 };
 
 function pointOnLine(p1, p2, length) {
+
+  if(!p2)
+    return p1
+
   var fullLength = Math.sqrt(Math.pow(p2[0] - p1[0], 2) + Math.pow(p2[1] - p1[1], 2));
   var n = fullLength - length;
 
